File: AEDS2/Aulai/estudos/fila.java
package Aulão.estudos;

 class celula {
    public int elemento;
    public celula prox;
    public celula(){
        this(0);
    }
    public celula(int x){
     this.elemento=x;
     this.prox=null;
    }


public class fila{
    public celula inicio;
    public celula fim;


    public fila(int x){
        fim=new celula (x);
        inicio=fim;
    }

    public void inserir(int x){
        fim.prox=new celula(x);
        fim= fim.prox;
    }

    public int remover() throws Exception{
        if(inicio==fim){
            throw new Exception("erro");
        celula tmp= inicio;
        int elemento=inicio.elemento;
        inicio=inicio.prox;
        tmp.prox=null;
        tmp=null;
        return elemento;
    }
        }
    }
}

File: AEDS2/Aulai/fila.c
#include <stdio.h>
#include <stdlib.h>

typedef struct celula
{
   int elemento;
   celula *prox;
}celula;

    celula *novacelula(int x){
       celula *nova = (celula*)malloc(sizeof (celula));
       nova->elemento=x;
       nova->prox=NULL;
       return nova;
    }


typedef struct fila
{
   celula *inicio;
   celula *fim;
}fila;
   
    void start(fila *f){
        f->inicio =novacelula(-1);
        f->fim=f->inicio;
    }

    void inserir(fila *f, int x){
        f->fim->prox= novacelula(x);
        f->fim=f->fim->prox;
    }
    int remover(fila *f){
        celula *tmp= f->inicio;
        int elemento= f->inicio->elemento;
        f->inicio= f->inicio->prox;
        tmp->prox=NULL;
        tmp=NULL;
        free(tmp);
        return elemento;
    }
    void mostrar(fila *f){
        celula *i;
        for(i=f->inicio; i!=NULL; i=i->prox){
            printf("%d", i->elemento );

        }
    }


